subsys/bluetooth/audio/cap_commander.c

```diff
@@ -77,18 +81,253 @@ int bt_cap_commander_discover(struct bt_conn *conn)
+	struct bt_bap_broadcast_assistant_add_src_param add_src_param = {0};
+
+	LOG_DBG("conn %p", (void *)conn);
+
+	if (!bt_cap_common_conn_in_active_proc(conn)) {
+
+		/* State change happened outside of a procedure; ignore */
+		return;
+	}
+
+	/* TODO We may have an issue with the proc_initated_cnt and proc_done_cnt not being in sync
+	 */
+	if (err != 0) {
+		LOG_DBG("Failed to add source: %d", err);
+		LOG_DBG("Aborting the proc %d %d", active_proc->proc_done_cnt,
+			active_proc->proc_initiated_cnt);
+
+		bt_cap_common_abort_proc(conn, err);
+	} else {
+		active_proc->proc_done_cnt++;
+
+		LOG_DBG("Conn %p broadcast source added (%zu/%zu streams done)", (void *)conn,
+			active_proc->proc_done_cnt, active_proc->proc_cnt);
+	}
+
+	if (bt_cap_common_proc_is_aborted()) {
+		if (bt_cap_common_proc_all_handled()) {
+			cap_commander_proc_complete();
+		}
+
+		return;
+	}
+
+	if (!bt_cap_common_proc_is_done()) {
+		const struct bt_cap_commander_proc_param *proc_param;
+
+		proc_param = &active_proc->proc_param.commander[active_proc->proc_done_cnt];
+		conn = proc_param->conn;
+		bt_addr_le_copy(&add_src_param.addr, &proc_param->broadcast_reception_start.addr);
+		add_src_param.adv_sid = proc_param->broadcast_reception_start.adv_sid;
+		add_src_param.broadcast_id = proc_param->broadcast_reception_start.broadcast_id;
+		add_src_param.pa_interval = proc_param->broadcast_reception_start.pa_interval;
+		add_src_param.num_subgroups = proc_param->broadcast_reception_start.num_subgroups;
+		add_src_param.subgroups = proc_param->broadcast_reception_start.subgroups;
+
+		active_proc->proc_initiated_cnt++;
+		err = bt_bap_broadcast_assistant_add_src(conn, &add_src_param);
+		if (err != 0) {
+			LOG_DBG("Failed to perform broadcast reception start for conn %p: %d",
+				(void *)conn, err);
+			bt_cap_common_abort_proc(conn, err);
+			cap_commander_proc_complete();
+		}
+	} else {
+		cap_commander_proc_complete();
+	}
+}
+
+static int cap_commander_register_ba_cb(void)
+{
+	int err = 0;
+
+	bt_bap_broadcast_assistant_register_cb(&broadcast_assistant_cb);
+	if (err != 0) {
+		LOG_DBG("Failed to register broadcast assistant callbacks: %d", err);
+
+		return -ENOEXEC;
+	}
+
+	ba_cb_registered = true;
+
+	return 0;
+}
+
+static bool valid_broadcast_reception_start_param(
+	const struct bt_cap_commander_broadcast_reception_start_param *param)
+{
+	CHECKIF(param == NULL) {
+		LOG_DBG("param is NULL");
+		return false;
+	}
+
+	CHECKIF(param->count == 0) {
+		LOG_DBG("Invalid param->count: %u", param->count);
+		return false;
+	}
+
+	CHECKIF(param->count > CONFIG_BT_MAX_CONN) {
+		LOG_DBG("param->count (%zu) is larger than CONFIG_BT_MAX_CONN (%d)", param->count,
+			CONFIG_BT_MAX_CONN);
+		return false;
+	}
+
+	CHECKIF(param->param == NULL) {
+		LOG_DBG("param->param is NULL");
+		return false;
+	}
+
+	for (size_t i = 0; i < param->count; i++) {
+		const struct bt_cap_commander_broadcast_reception_start_member_param *start_param =
```

Comment:
Several values of this is not being properly checked for valid values:
`addr`, `member`, `adv_sid`, `pa_interval` and `broadcast_id`